src/fairseq2/data/data_pipeline.py

```diff
@@ -231,6 +231,35 @@ def collate(
             See :py:class:`fairseq2.data.Collater` for details.
             """
 
+        def dynamic_bucket(
+            self,
+            threshold: float,
+            cost_fn: Callable[[Any], float],
```

Comment:
i think we may add an extra check that makes sure that `cost_fn` function returns only non-negative scores.